storage/tools/dump_tree/main.go

```diff
@@ -195,7 +203,42 @@ func main() {
 	}
 
 	// All leaves are now sequenced into the tree. The current state is what we need.
-	memory.DumpSubtrees(ls, tree.TreeId, func(k string, v *storagepb.SubtreeProto) {
-		fmt.Printf("%s: %s\n", k, v.String())
-	})
+	glog.Info("Producing output")
+
+	if *latestRevisionFlag {
+		vMap := make(map[string]treeandrev)
+
+		memory.DumpSubtrees(ls, tree.TreeId, func(k string, v *storagepb.SubtreeProto) {
+			// Relies on the btree key space for subtrees being /tree_id/subtree/<id>/<revision>
+			pieces := strings.Split(k, "/")
+			if got, want := len(pieces), 5; got != want {
+				glog.Fatalf("Wrong no of Btree subtree key segments. Got: %d, want: %d", got, want)
+			}
+
+			e := vMap[pieces[3]]
```

Comment:
might be worth pulling these pieces out to named vars to make it easier to read?